### ISSUE TITLE
How to Update Model in Built_redux ?

### ISSUE THREAD

**Issue by ejabu** [brianegan/flutter_architecture_samples#61]
I still don't understand how to use this one ?

for example.. I add date value in Todo Model.

```

  factory Todo(String task) {
    return _$Todo._(
      task: task,
      complete: false,
      note: '',
      date: '',
      id: Uuid().generateV4(),
    );
  }

  factory Todo.builder([updates(TodoBuilder b)]) {
    final builder = TodoBuilder()
      ..id = Uuid().generateV4()
      ..complete = false
      ..note = ''
      ..date = ''
      ..update(updates);

    return builder.build();
  }



```

But I got this

```
The named parameter 'date' isn't defined.
The setter 'date' isn't defined for the class 'TodoBuilder'.
Missing concrete implementation of getter 'Todo.date'.
```

**I read that it has something to do with built_value 

but I already run this on my terminal after editting the code and while running the app on my Phone**

```
flutter packages pub run build_runner build
[INFO] Generating build script...
[INFO] Generating build script completed, took 338ms

```

How to properly setup the workspace, so the file with *.g.dart extension can be updated as soon as I develop the project ?

**Comment by mmcc007** [comment]
have u tried
```bash
flutter packages pub run build_runner watch
```
If you leave this running on the command line and while adding fields to a correctly configured model, it will detect the changes to the model and rebuild the related *.g.dart on the fly.

**Comment by ejabu** [comment]
Thanks, I tried and this is The Output


```

[INFO] Generating build script...
[INFO] Generating build script completed, took 313ms

[INFO] Setting up file watchers...
[INFO] Setting up file watchers completed, took 26ms

[INFO] Waiting for all file watchers to be ready...
[INFO] Waiting for all file watchers to be ready completed, took 204ms

[INFO] Initializing inputs
[INFO] Building new asset graph...


You have hit a bug in build_runner
Please file an issue with reproduction steps at https://github.com/dart-lang/build/issues


DuplicateAssetNodeError: GeneratedAssetNode: built_redux_sample|lib/main.g.dart generated from input built_redux_sample|lib/main.dart.
This probably means you have multiple actions that are trying to output the same file.
package:build_runner/src/asset_graph/graph.dart 462:11          AssetGraph._addGeneratedOutputs
package:build_runner/src/asset_graph/graph.dart 414:21          AssetGraph._addInBuildPhaseOutputs
package:build_runner/src/asset_graph/graph.dart 380:26          AssetGraph._addOutputsForSources
package:build_runner/src/asset_graph/graph.dart 55:11           AssetGraph.build
package:build_runner/src/generate/build_definition.dart 110:39  _Loader.prepareWorkspace.<fn>
package:build_runner/src/logging/logging.dart 25:30             logTimedAsync
package:build_runner/src/generate/build_definition.dart 109:13  _Loader.prepareWorkspace
package:build_runner/src/generate/build_definition.dart 65:12   BuildDefinition.prepareWorkspace
package:build_runner/src/generate/watch_impl.dart 267:48        WatchImpl._run.<fn>
package:build_runner/src/generate/watch_impl.dart 289:6         WatchImpl._run
package:build_runner/src/generate/watch_impl.dart 162:9         new WatchImpl
package:build_runner/src/generate/watch_impl.dart 117:9         runWatch
package:build_runner/src/generate/watch_impl.dart 96:7          watch
package:build_runner/src/generate/build.dart 132:5              watch
package:build_runner/src/entrypoint/options.dart 364:25         _WatchCommand.run
package:args/command_runner.dart 194:27                         CommandRunner.runCommand
package:args/command_runner.dart 109:29                         CommandRunner.run.<fn>
dart:async                                                      new Future.sync
package:args/command_runner.dart 109:11                         CommandRunner.run
package:build_runner/src/entrypoint/run.dart 22:31              run
.dart_tool/build/entrypoint/build.dart 15:22                    main

pub finished with exit code 1


```

**Comment by mmcc007** [comment]
There might well be a bug. When I tried it complained that the generated files already existed (`UnexpectedExistingOutputsException`). I got around it with
```
flutter packages pub run build_runner watch --delete-conflicting-outputs
```
BTW: I was on the beta channel when doing this.

In your case the error is suggesting that there is an existing process already running that is trying to output the same files. You might want to check for and kill other dart processes that are doing this.

BTW: I assume you have declared the date in the Todo class something like:
``` dart
String get date;
```
or
```dart
DateTime get date;
```
